src/lib.rs

```diff
@@ -1076,6 +1115,60 @@ impl GPT {
             .ok_or(Error::NoSpaceLeft)
     }
 
+    /// Get the range of bytes covered by a partition.
+    ///
+    /// # Errors
+    ///
+    /// This function will return an error if the partition number is invalid, or if
+    /// the partition is unused, or if the partition's `ending_lba` is less than its
+    /// `starting_lba`.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// let ss = 512;
+    /// let data = vec![0; 100 * ss as usize];
+    /// let mut cur = std::io::Cursor::new(data);
+    /// let mut gpt = gptman::GPT::new_from(&mut cur, ss as u64, [0xff; 16])
+    ///     .expect("could not create partition table");
+    /// gpt[1] = gptman::GPTPartitionEntry {
+    ///     partition_type_guid: [0xff; 16],
+    ///     unique_partition_guid: [0xff; 16],
+    ///     starting_lba: 2048,
+    ///     ending_lba: 2048,
+    ///     attribute_bits: 0,
+    ///     partition_name: "A Robot Named Fight!".into(),
+    /// };
+    ///
+    /// assert_eq!(gpt.get_partition_byte_range(1).unwrap(), 1048576..=1049087);
+    /// ```
+    pub fn get_partition_byte_range(&self, partition_number: u32) -> Result<RangeInclusive<u64>> {
+        if partition_number == 0 || partition_number > self.header.number_of_partition_entries {
+            return Err(Error::InvalidPartitionNumber(partition_number));
+        }
+        let partition = &self[partition_number];
+        if partition.is_unused() {
+            return Err(Error::UnusedPartition(partition_number));
+        }
+
+        let sector_range = partition.range()?;
+
+        let start_byte = sector_range
+            .start()
+            .checked_mul(self.sector_size)
+            .ok_or(Error::Overflow)?;
+        // Sector ranges are inclusive, so to get the position of the end byte we need
+        // to add the size of another sector, less one byte because the byte range
+        // returned from this function is also inclusive.
+        let end_byte = sector_range
+            .end()
+            .checked_mul(self.sector_size)
+            .and_then(|v| v.checked_add(self.sector_size))
+            .and_then(|v| v.checked_sub(1))
```

Comment:
I guess we could remove one overflow check like this.

```suggestion
            .and_then(|v| v.checked_add(self.sector_size - 1))
```

Reply:
Done